client/lib/mobile-app/index.js

```diff
@@ -0,0 +1,10 @@
+/** @format **/
+
+/**
+ * Returns whether user is using a mobile app.
+ *
+ * @returns {Boolean} Whether the user agent matches the ones used on the mobile apps.
+ */
+export function isMobileApp() {
+	return navigator && /wp-(android|iphone)/.test( navigator.userAgent );
```

Comment:
I don't know if we actually have to check for `navigator`. `AppBanner` does this:
`typeof window !== 'undefined' ? navigator.userAgent : ''`, so maybe a `window` check makes more sense here?

Reply:
The `window` is implied by the browser I believe; checking `navigator` is the same as checking `window.navigator`.

Reply:
I think it's safe to just check `navigator` for what @kwight said. We also have some other places where we assume `navigator` is available ([#](https://github.com/Automattic/wp-calypso/blob/2b34410c2b0bcc828455b290f9797e35978dffcf/client/my-sites/site-settings/form-writing.jsx#L36), [#](https://github.com/Automattic/wp-calypso/blob/2b34410c2b0bcc828455b290f9797e35978dffcf/client/components/tinymce/plugins/wpcom-view/plugin.js#L41)).